uniffi_bindgen/src/bindings/kotlin/gen_kotlin/primitives.rs

```diff
@@ -2,373 +2,158 @@
-        return this.toInt()
-    }
-
-    internal fun UInt.write(buf: RustBufferBuilder) {
-        buf.putInt(this.toInt())
-    }
-"#
-);
-
-impl_code_type_for_primitive!(
-    UInt64CodeType,
-    "ULong",
-    r#"
-    internal fun ULong.Companion.lift(v: Long): ULong {
-        return v.toULong()
-    }
-
-    internal fun ULong.Companion.read(buf: ByteBuffer): ULong {
-        return ULong.lift(buf.getLong())
-    }
-
-    internal fun ULong.lower(): Long {
-        return this.toLong()
-    }
-
-    internal fun ULong.write(buf: RustBufferBuilder) {
-        buf.putLong(this.toLong())
-    }
-"#
-);
-
-impl_code_type_for_primitive!(
-    Float32CodeType,
-    "Float",
-    r#"
-    internal fun Float.Companion.lift(v: Float): Float {
-        return v
-    }
-
-    internal fun Float.Companion.read(buf: ByteBuffer): Float {
-        return buf.getFloat()
-    }
-
-    internal fun Float.lower(): Float {
-        return this
-    }
+                Type::Float32 => format!("{}f", num_str),
+                Type::Float64 => num_str,
+                _ => panic!("Unexpected literal: {} is not a number", num_str),
+            }
+        }
 
-    internal fun Float.write(buf: RustBufferBuilder) {
-        buf.putFloat(this)
+        match literal {
+            Literal::Boolean(v) => format!("{}", v),
+            Literal::String(s) => format!("\"{}\"", s),
+            Literal::Int(i, radix, type_) => typed_number(
+                &type_,
+                match radix {
+                    Radix::Octal => format!("{:#x}", i),
+                    Radix::Decimal => format!("{}", i),
+                    Radix::Hexadecimal => format!("{:#x}", i),
+                },
+            ),
+            Literal::UInt(i, radix, type_) => typed_number(
+                &type_,
+                match radix {
+                    Radix::Octal => format!("{:#x}", i),
+                    Radix::Decimal => format!("{}", i),
+                    Radix::Hexadecimal => format!("{:#x}", i),
+                },
+            ),
+            Literal::Float(string, type_) => typed_number(&type_, string.clone()),
+
+            _ => unreachable!("Literal"),
+        }
     }
-"#
-);
+}
 
-impl_code_type_for_primitive!(
-    Float64CodeType,
-    "Double",
-    r#"
-    internal fun Double.Companion.lift(v: Double): Double {
-        return v
-    }
+#[derive(Template, Hash)]
+#[template(syntax = "kt", escape = "none", path = "String.kt")]
+struct StringTemplate;
+
+#[derive(Template, Hash)]
+#[template(syntax = "kt", escape = "none", path = "Primitive.kt")]
+struct PrimitiveTemplate {
+    type_name: String,
+    ffi_name: String,
+    lift_expr: String,
+    lower_expr: String,
+    read_expr: String,
+    write_expr: String,
+}
 
-    internal fun Double.Companion.read(buf: ByteBuffer): Double {
-        val v = buf.getDouble()
-        return v
+impl PrimitiveTemplate {
+    fn int(handler: &SimpleTypeHandler, get_func: &str) -> Self {
```

Comment:
It's a bit jarring to see this switching back and forth between "simple type" and "primitive template", what do you think about trying to consolidate on "primitive type" as the name for these things throughout the codebase?